client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js

```diff
@@ -0,0 +1,448 @@
+	}
+
+	woocommerceServiceLinkClicked = () => {
+		this.trackElementClicked( 'shipping_banner_woocommerce_service_link' );
+	};
+
+	trackBannerEvent = ( eventName, customProps = {} ) => {
+		const { activePlugins, isJetpackConnected, wcsPluginSlug } = this.props;
+		recordEvent( eventName, {
+			banner_name: 'wcadmin_install_wcs_prompt',
+			jetpack_installed: activePlugins.includes( 'jetpack' ),
+			jetpack_connected: isJetpackConnected,
+			wcs_installed: activePlugins.includes( wcsPluginSlug ),
+			...customProps,
+		} );
+	};
+
+	trackImpression = () => {
+		this.trackBannerEvent( 'banner_impression' );
+	};
+
+	trackElementClicked = ( element ) => {
+		this.trackBannerEvent( 'banner_element_clicked', {
+			element,
+		} );
+	};
+
+	acceptTosAndGetWCSAssets() {
+		return acceptWcsTos()
+			.then( () => getWcsAssets() )
+			.then( ( wcsAssets ) => this.loadWcsAssets( wcsAssets ) )
+			.catch( () => this.setState( { wcsSetupError: true } ) );
+	}
+
+	generateMetaBoxHtml( title, args ) {
+		const argsJsonString = JSON.stringify( args ).replace( /"/g, '&quot;' ); // JS has no native html_entities so we just replace.
+
+		const togglePanelText = __( 'Toggle panel:', 'woocommerce-admin' );
+
+		return `
+<div id="woocommerce-order-label" class="postbox ">
+	<button type="button" class="handlediv" aria-expanded="true">
+		<span class="screen-reader-text">${ togglePanelText } ${ title }</span>
+		<span class="toggle-indicator" aria-hidden="true"></span>
+	</button>
+	<h2 class="hndle"><span>${ title }</span></h2>
+	<div class="inside">
+		<div class="wcc-root woocommerce wc-connect-create-shipping-label" data-args="${ argsJsonString }">
+		</div>
+	</div>
+</div>
+`;
+	}
+
+	loadWcsAssets( { assets } ) {
+		if ( this.state.wcsAssetsLoaded || this.state.wcsAssetsLoading ) {
+			this.openWcsModal();
+			return;
+		}
+
+		this.setState( { wcsAssetsLoading: true } );
+
+		const js = assets.wc_connect_admin_script;
+		const styles = assets.wc_connect_admin_style;
+
+		const { orderId } = this.state;
+		const { itemsCount } = this.props;
+
+		const shippingLabelContainerHtml = this.generateMetaBoxHtml(
+			__( 'Shipping Label', 'woocommerce-admin' ),
+			{
+				orderId,
+				context: 'shipping_label',
+				items: itemsCount,
+			}
+		);
+		// Insert shipping label metabox just above main order details box.
+		document
+			.getElementById( 'woocommerce-order-data' )
+			.insertAdjacentHTML( 'beforebegin', shippingLabelContainerHtml );
+
+		const shipmentTrackingHtml = this.generateMetaBoxHtml(
+			__( 'Shipment Tracking', 'woocommerce-admin' ),
+			{
+				orderId,
+				context: 'shipment_tracking',
+				items: itemsCount,
+			}
+		);
+		// Insert tracking metabox in the side after the order actions.
+		document
+			.getElementById( 'woocommerce-order-actions' )
+			.insertAdjacentHTML( 'afterend', shipmentTrackingHtml );
+
+		// Need to refresh so the new metaboxes are sortable.
+		window.jQuery( '#normal-sortables' ).sortable( 'refresh' );
```

Comment:
Perhaps my optimism here is too great, but should we wrap these lines in an `if ( window.jQuery )` for the maybe someday when it will no longer be present?

Reply:
Can't hurt